identity-account-storage/src/storage/traits.rs

```diff
@@ -27,45 +28,82 @@ mod storage_sub_trait {
 
-  /// Inserts a private key at the specified `location` and returns its `PublicKey`.
-  async fn key_insert(&self, did: &IotaDID, location: &KeyLocation, private_key: PrivateKey) -> Result<PublicKey>;
+  /// Moves a key from `source` to `target`.
+  ///
+  /// The key at the source location is removed. If a key at the `target` exists, it is overwritten.
+  async fn key_move(&self, account_id: &AccountId, source: &KeyLocation, target: &KeyLocation) -> Result<()>;
 
-  /// Retrieves the public key at the specified `location`.
-  async fn key_get(&self, did: &IotaDID, location: &KeyLocation) -> Result<PublicKey>;
+  /// Retrieves the public key from `location`.
+  async fn key_public(&self, account_id: &AccountId, location: &KeyLocation) -> Result<PublicKey>;
 
-  /// Deletes the keypair specified by `location`.
-  async fn key_del(&self, did: &IotaDID, location: &KeyLocation) -> Result<()>;
+  /// Deletes the key at `location`.
+  ///
+  /// This operation is idempotent: it does not fail if the key does not exist.
+  async fn key_del(&self, account_id: &AccountId, location: &KeyLocation) -> Result<()>;
 
   /// Signs `data` with the private key at the specified `location`.
-  async fn key_sign(&self, did: &IotaDID, location: &KeyLocation, data: Vec<u8>) -> Result<Signature>;
+  async fn key_sign(&self, account_id: &AccountId, location: &KeyLocation, data: Vec<u8>) -> Result<Signature>;
 
-  /// Returns `true` if a keypair exists at the specified `location`.
-  async fn key_exists(&self, did: &IotaDID, location: &KeyLocation) -> Result<bool>;
+  /// Returns `true` if a key exists at the specified `location`.
+  async fn key_exists(&self, account_id: &AccountId, location: &KeyLocation) -> Result<bool>;
 
-  /// Returns the chain state of the identity specified by `did`.
-  async fn chain_state(&self, did: &IotaDID) -> Result<Option<ChainState>>;
+  /// Returns the chain state of the identity specified by `account_id`.
+  async fn chain_state(&self, account_id: &AccountId) -> Result<Option<ChainState>>;
 
-  /// Set the chain state of the identity specified by `did`.
-  async fn set_chain_state(&self, did: &IotaDID, chain_state: &ChainState) -> Result<()>;
+  /// Set the chain state of the identity specified by `account_id`.
+  async fn set_chain_state(&self, account_id: &AccountId, chain_state: &ChainState) -> Result<()>;
 
-  /// Returns the state of the identity specified by `did`.
-  async fn state(&self, did: &IotaDID) -> Result<Option<IdentityState>>;
+  /// Returns the [`IotaDocument`] of the identity specified by `account_id`.
+  async fn document(&self, account_id: &AccountId) -> Result<Option<IotaDocument>>;
 
-  /// Sets a new state for the identity specified by `did`.
-  async fn set_state(&self, did: &IotaDID, state: &IdentityState) -> Result<()>;
+  /// Sets a new state for the identity specified by `account_id`.
+  async fn set_document(&self, account_id: &AccountId, state: &IotaDocument) -> Result<()>;
```

Comment:
What naming scheme are we following in the `Storage`? We have `noun_verb` elsewhere like `key_insert`, `key_sign`, `index_set`, `index_get` but here we have `verb_noun` for `set_chain_state` and `set_document`. Should we be consistent?

Reply:
I would make the state data structure consistent with the rest, so e.g. `document_set`. Should we even use `chain_state_get` and `document_get`, that feels maximally consistent?

Reply:
Yeah I would change all functions in `Storage` to be `noun_verb` in then, as long as it's internally consistent, even though it's inconsistent with the rest of the project. Rust and Wasm function names should always match as much as possible.